### PR TITLE
Revert 9273

### PR DESCRIPTION
**Pull request by Dref360**
This PR revert #9273 

This PR is not backward compatible with most scripts so we should revert it and think of a better way.

### CODE REVIEW

**Comment by Vijayabhaskar96** [comment]
I guess the changes where not properly reverted, I get this error now,or the problem is with my code?
`Epoch 1/10
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
/usr/local/lib/python3.6/dist-packages/keras/utils/data_utils.py in get(self)
    577             while self.is_running():
--> 578                 inputs = self.queue.get(block=True).get()
    579                 self.queue.task_done()

/usr/lib/python3.6/multiprocessing/pool.py in get(self, timeout)
    643         else:
--> 644             raise self._value
    645 

/usr/lib/python3.6/multiprocessing/pool.py in worker(inqueue, outqueue, initializer, initargs, maxtasks, wrap_exception)
    118         try:
--> 119             result = (True, func(*args, **kwds))
    120         except Exception as e:

/usr/local/lib/python3.6/dist-packages/keras/utils/data_utils.py in get_index(uid, i)
    400     """
--> 401     return _SHARED_SEQUENCES[uid][i]
    402 

/usr/local/lib/python3.6/dist-packages/keras/preprocessing/image.py in __getitem__(self, idx)
    826                                        self.batch_size * (idx + 1)]
--> 827         return self._get_batches_of_transformed_samples(index_array)
    828 

/usr/local/lib/python3.6/dist-packages/keras/preprocessing/image.py in _get_batches_of_transformed_samples(self, index_array)
   1233             x = self.image_data_generator.random_transform(x)
-> 1234             x = self.image_data_generator.standardize(x)
   1235             batch_x[i] = x

/usr/local/lib/python3.6/dist-packages/keras/preprocessing/image.py in standardize(self, x)
    583         if self.preprocessing_function:
--> 584             x = self.image_data_generator.preprocessing_function(x)
    585         if self.rescale:

AttributeError: 'ImageDataGenerator' object has no attribute 'image_data_generator'

The above exception was the direct cause of the following exception:

StopIteration                             Traceback (most recent call last)
<ipython-input-18-206f186a346e> in <module>()
      4                                   callbacks=[c1,c2],
      5                                   validation_steps=STEP_SIZE_VALID,
----> 6                                   epochs=10)

/usr/local/lib/python3.6/dist-packages/keras/legacy/interfaces.py in wrapper(*args, **kwargs)
     89                 warnings.warn('Update your `' + object_name +
     90                               '` call to the Keras 2 API: ' + signature, stacklevel=2)
---> 91             return func(*args, **kwargs)
     92         wrapper._original_function = func
     93         return wrapper

/usr/local/lib/python3.6/dist-packages/keras/engine/training.py in fit_generator(self, generator, steps_per_epoch, epochs, verbose, callbacks, validation_data, validation_steps, class_weight, max_queue_size, workers, use_multiprocessing, shuffle, initial_epoch)
   2190                 batch_index = 0
   2191                 while steps_done < steps_per_epoch:
-> 2192                     generator_output = next(output_generator)
   2193 
   2194                     if not hasattr(generator_output, '__len__'):

/usr/local/lib/python3.6/dist-packages/keras/utils/data_utils.py in get(self)
    582         except Exception as e:
    583             self.stop()
--> 584             six.raise_from(StopIteration(e), e)
    585 
    586     def _send_sequence(self):

/usr/local/lib/python3.6/dist-packages/six.py in raise_from(value, from_value)

StopIteration: 'ImageDataGenerator' object has no attribute 'image_data_generator'`

File: keras/preprocessing/image.py [hunk 1]
```diff
@@ -580,6 +580,8 @@ def standardize(self, x):
         # Returns
             The inputs, normalized.
         """
+        if self.preprocessing_function:
```
Comment (datumbox) [inline keras/preprocessing/image.py]: @Dref360 Minor typo. This needs to be:
```
if self.preprocessing_function:
      x = self.preprocessing_function(x)
```

**Comment by datumbox** [comment]
@Dref360 @fchollet Any chance for a hot fix release? The latest stable makes it impossible using the pre-trained models or doing transfer learning. Thanks!

File: keras/preprocessing/image.py (hunk 1 above)
Comment (datumbox) [reply keras/preprocessing/image.py]: @Dref360 Might be also worth changing the documentation of preprocessing_function from "The function will run before any other modification on it." to "The function will run before any other standardization is applied." or something similar. :)

**Comment by Vijayabhaskar96** [comment]
I noticed that typo too, but I don't want to submit a PR to fix this, because there may be some other code that may be left out being reverted as this typo, anyway I fixed the typo and works fine at my end. If anybody wants a quick fix just `pip install git+https://github.com/Vijayabhaskar96/keras.git`

File: keras/preprocessing/image.py (hunk 1 above)
Comment (Dref360) [reply keras/preprocessing/image.py]: My bad sorry :'( We didn't have any test for that (weird...) So I added a test, will get merged shortly.

Comment (Dref360) [reply keras/preprocessing/image.py]: As for the doc, We'll handle it later.

Comment (datumbox) [reply keras/preprocessing/image.py]: sounds good thanks!

**Pull request merged**